domain/rules/level-three/bleu-rouge-rule.ts

```diff
@@ -0,0 +1,99 @@
+import { DiceRoll } from "../dice-rule";
+import { DiceRollGameContext } from "../../game-context-event";
+import { RuleEffect, RuleEffectEvent, RuleEffects } from "../rule-effect";
+import { Resolver } from "../rule-resolver";
+import { ChouetteRule } from "../basic-rules/chouette-rule";
+
+export interface BleuRougeResolution {
+  diceRoll: DiceRoll;
+  bids: Array<BleuRougeBid>;
+}
+
+export class BleuRougeRule extends ChouetteRule {
+  constructor(private readonly resolver: Resolver<BleuRougeResolution>) {
+    super();
+  }
+
+  isApplicableToDiceRoll(diceRoll: DiceRoll): boolean {
+    const [dieValue1, dieValue2, dieValue3] = [...diceRoll].sort();
+
+    return dieValue1 === 3 && dieValue2 === 3 && dieValue3 === 4;
+  }
+
+  async applyDiceRule(context: DiceRollGameContext): Promise<RuleEffects> {
+    const [chouetteRuleEffect] = await super.applyDiceRule(context);
+
+    const initialBleuRougeRulEffect: RuleEffect = {
+      ...chouetteRuleEffect,
+      event: RuleEffectEvent.BLEU_ROUGE,
+    };
+
+    const { bids, diceRoll } = await this.resolver.getResolution();
+
+    const addJarretRuleEffects: RuleEffects = [];
+    const secondRollContext: DiceRollGameContext = { ...context, diceRoll };
+    if (context.runner.isDiceRollANeant(secondRollContext)) {
+      addJarretRuleEffects.push({
+        event: RuleEffectEvent.ADD_JARRET,
+        playerName: context.playerName,
+        score: 0,
+      });
+    }
+
+    const diceRollSum = diceRoll.reduce((sum, dieValue) => sum + dieValue, 0);
```

Comment:
```suggestion
    const diceRollSum = diceRoll[0] + diceRoll[1] + diceRoll[2];
```

Reply:
![reduce-ew](https://user-images.githubusercontent.com/6702319/99645371-67f7ad00-2a4f-11eb-8bf8-2b893ae4e739.jpg)

Reply:
> Keep it Simple, Stupid...

I think the sum without reduce is much more "simple, stupid"